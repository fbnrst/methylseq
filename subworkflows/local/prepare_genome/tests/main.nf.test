nextflow_workflow {

    name "Test Workflow PREPARE_GENOME"
    script "../main.nf"
    workflow "PREPARE_GENOME"

    tag "subworkflows/local"
    tag "prepare_genome"
    tag "bismark/genomepreparation"
    tag "bwameth/index"
    tag "samtools/faidx"

    test("Bismark") {

        when {
            params {
                aligner = "bismark"
            }
            workflow {
                """
                
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
          )
        }

    }

    test("Bwameth") {

        when {
            params {
                aligner = "bwameth"
            }
            workflow {
                """
                
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
          )
        }

    }

}
