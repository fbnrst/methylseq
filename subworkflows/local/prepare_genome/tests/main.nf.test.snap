{
    "Bismark": {
        "content": [
            {
                "0": [
                    "/nf-core/test-datasets/methylseq/reference/genome.fa"
                ],
                "1": [
                    [
                        [
                            [
                                "BS_CT.1.bt2:md5,d53bef7d951c7d08e327944581c911eb",
                                "BS_CT.2.bt2:md5,752a03fc364115fc910e5a28ae154382",
                                "BS_CT.3.bt2:md5,b41e72eefad74e8ede411c78a5bd5dee",
                                "BS_CT.4.bt2:md5,02b3d1855a67fd9cbb4c411406a22fde",
                                "BS_CT.rev.1.bt2:md5,52ae603fff473b3b25eecd36c533edb9",
                                "BS_CT.rev.2.bt2:md5,0e290e05f7dae065cae54e29ed97afe5",
                                "genome_mfa.CT_conversion.fa:md5,4dd181b65f1f3549f4132c9a3759eee8"
                            ],
                            [
                                "BS_GA.1.bt2:md5,9aff51b1712758b891b0c427a988977f",
                                "BS_GA.2.bt2:md5,579241d95941538b2e75bbdb6cfaa73d",
                                "BS_GA.3.bt2:md5,b41e72eefad74e8ede411c78a5bd5dee",
                                "BS_GA.4.bt2:md5,6b64fa496ffa2af2af235e1c87f06a25",
                                "BS_GA.rev.1.bt2:md5,dabf2d6f80ba7e8fc96dfdc203f6bbfd",
                                "BS_GA.rev.2.bt2:md5,fac2f8b39b0e15dec7aa79a0ecf12b49",
                                "genome_mfa.GA_conversion.fa:md5,3dee732bc344fd620c3a7322f52c6a5f"
                            ]
                        ],
                        "genome.fa:md5,923a0a268ad29fee3c3437d00f9970de"
                    ]
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    "versions.yml:md5,48a4baa99421c3dd839946dd1af8d7aa"
                ],
                "bismark_index": [
                    [
                        [
                            [
                                "BS_CT.1.bt2:md5,d53bef7d951c7d08e327944581c911eb",
                                "BS_CT.2.bt2:md5,752a03fc364115fc910e5a28ae154382",
                                "BS_CT.3.bt2:md5,b41e72eefad74e8ede411c78a5bd5dee",
                                "BS_CT.4.bt2:md5,02b3d1855a67fd9cbb4c411406a22fde",
                                "BS_CT.rev.1.bt2:md5,52ae603fff473b3b25eecd36c533edb9",
                                "BS_CT.rev.2.bt2:md5,0e290e05f7dae065cae54e29ed97afe5",
                                "genome_mfa.CT_conversion.fa:md5,4dd181b65f1f3549f4132c9a3759eee8"
                            ],
                            [
                                "BS_GA.1.bt2:md5,9aff51b1712758b891b0c427a988977f",
                                "BS_GA.2.bt2:md5,579241d95941538b2e75bbdb6cfaa73d",
                                "BS_GA.3.bt2:md5,b41e72eefad74e8ede411c78a5bd5dee",
                                "BS_GA.4.bt2:md5,6b64fa496ffa2af2af235e1c87f06a25",
                                "BS_GA.rev.1.bt2:md5,dabf2d6f80ba7e8fc96dfdc203f6bbfd",
                                "BS_GA.rev.2.bt2:md5,fac2f8b39b0e15dec7aa79a0ecf12b49",
                                "genome_mfa.GA_conversion.fa:md5,3dee732bc344fd620c3a7322f52c6a5f"
                            ]
                        ],
                        "genome.fa:md5,923a0a268ad29fee3c3437d00f9970de"
                    ]
                ],
                "bwameth_index": [
                    
                ],
                "fasta": [
                    "/nf-core/test-datasets/methylseq/reference/genome.fa"
                ],
                "fasta_index": [
                    
                ],
                "versions": [
                    "versions.yml:md5,48a4baa99421c3dd839946dd1af8d7aa"
                ]
            }
        ],
        "timestamp": "2023-12-17T14:43:46.219838806"
    },
    "Bwameth": {
        "content": [
            {
                "0": [
                    "/nf-core/test-datasets/methylseq/reference/genome.fa"
                ],
                "1": [
                    
                ],
                "2": [
                    [
                        "genome.fa:md5,923a0a268ad29fee3c3437d00f9970de",
                        "genome.fa.bwameth.c2t:md5,e51d48ed28fa0c26e2f9c9f13d09403b",
                        "genome.fa.bwameth.c2t.amb:md5,010a242c6764efb30141868a45d698b3",
                        "genome.fa.bwameth.c2t.ann:md5,09b4db3d87a2d4dac9e10e807f377110",
                        "genome.fa.bwameth.c2t.bwt:md5,4646f8ae6bd523b7f4106bbd32eff95e",
                        "genome.fa.bwameth.c2t.pac:md5,a662d7add09698e25c8152fd2306ec66",
                        "genome.fa.bwameth.c2t.sa:md5,4a1e905b828396a1909669bc9c573d8d"
                    ]
                ],
                "3": [
                    "/nf-core/test-datasets/methylseq/reference/genome.fa.fai"
                ],
                "4": [
                    "versions.yml:md5,c53196f9c3908f996b30770ef3a60a00"
                ],
                "bismark_index": [
                    
                ],
                "bwameth_index": [
                    [
                        "genome.fa:md5,923a0a268ad29fee3c3437d00f9970de",
                        "genome.fa.bwameth.c2t:md5,e51d48ed28fa0c26e2f9c9f13d09403b",
                        "genome.fa.bwameth.c2t.amb:md5,010a242c6764efb30141868a45d698b3",
                        "genome.fa.bwameth.c2t.ann:md5,09b4db3d87a2d4dac9e10e807f377110",
                        "genome.fa.bwameth.c2t.bwt:md5,4646f8ae6bd523b7f4106bbd32eff95e",
                        "genome.fa.bwameth.c2t.pac:md5,a662d7add09698e25c8152fd2306ec66",
                        "genome.fa.bwameth.c2t.sa:md5,4a1e905b828396a1909669bc9c573d8d"
                    ]
                ],
                "fasta": [
                    "/nf-core/test-datasets/methylseq/reference/genome.fa"
                ],
                "fasta_index": [
                    "/nf-core/test-datasets/methylseq/reference/genome.fa.fai"
                ],
                "versions": [
                    "versions.yml:md5,c53196f9c3908f996b30770ef3a60a00"
                ]
            }
        ],
        "timestamp": "2023-12-17T14:43:53.866579307"
    }
}